public/pages/CreateMonitor/components/MonitorExpressions/expressions/utils/helpers.js

```diff
@@ -36,13 +36,33 @@ export function getOfExpressionAllowedTypes(values) {
 
 export function getMetricExpressionAllowedTypes(values) {
   const types = ['number'];
-  if (['min', 'max', 'count'].includes(values.aggregationType)) types.push('date');
-  if (['count'].includes(values.aggregationType)) types.push('keyword');
-
+  if (
+    values.aggregations.some((e) => {
+      return ['min', 'max', 'count'].includes(e.aggregationType);
+    })
+  )
+    types.push('date');
+  if (
+    values.aggregations.some((e) => {
+      return ['count'].includes(e.aggregationType);
+    })
+  )
+    types.push('keyword');
```

Comment:
Because of the big if statements, could we make it clear what the body of the if statements are?

Reply:
Added comments